pkg/autoscaler/autoscaler/util.go

```diff
@@ -137,6 +203,80 @@ func defaultTAC(tac *v1alpha1.TidbClusterAutoScaler) {
 	}
 }
 
+func validateBasicAutoScalerSpec(tac *v1alpha1.TidbClusterAutoScaler, spec *v1alpha1.BasicAutoScalerSpec, component v1alpha1.MemberType) error {
+	if len(spec.Rules) == 0 {
+		return fmt.Errorf("no rules defined for component %s in %s/%s", component.String(), tac.Namespace, tac.Name)
+	}
+
+	acceptableResources := map[corev1.ResourceName]struct{}{
+		corev1.ResourceCPU:     {},
+		corev1.ResourceStorage: {},
+	}
+
+	checkCommon := func(res corev1.ResourceName, rule v1alpha1.AutoRule) error {
+		if _, ok := acceptableResources[res]; !ok {
+			return fmt.Errorf("unknown resource type %s of %s in %s/%s", res.String(), component.String(), tac.Namespace, tac.Name)
+		}
+		if rule.MaxThreshold > 1.0 || rule.MaxThreshold < 0.0 {
+			return fmt.Errorf("max threshold should be between 0 and 1 for rule %s of %s in %s/%s", res, component.String(), tac.Namespace, tac.Name)
+		}
+		if rule.MinThreshold != nil && (*rule.MinThreshold > 1.0 || *rule.MinThreshold < 0.0) {
+			return fmt.Errorf("min threshold should be between 0 and 1 for rule %s of %s in %s/%s", res, component.String(), tac.Namespace, tac.Name)
+		}
+		for _, resType := range rule.ResourceTypes {
+			if _, ok := tac.Spec.Resources[resType]; !ok {
+				return fmt.Errorf("unknown resource %s for %s in %s/%s", resType, component.String(), tac.Namespace, tac.Name)
+			}
+		}
+		return nil
+	}
+
+	for res, rule := range spec.Rules {
+		if err := checkCommon(res, rule); err != nil {
+			return err
+		}
+
+		switch res {
+		case corev1.ResourceCPU:
+			if *rule.MinThreshold > rule.MaxThreshold {
+				return fmt.Errorf("min threshold > max threshold for CPU rule of %s in %s/%s", component.String(), tac.Namespace, tac.Name)
+			}
+		case corev1.ResourceStorage:
+			for _, resType := range rule.ResourceTypes {
+				if res, ok := tac.Spec.Resources[resType]; ok {
+					if res.Storage.Cmp(zeroQuantity) == 0 {
+						return fmt.Errorf("resource %s specified for %s in %s/%s does not have storage", resType, component.String(), tac.Namespace, tac.Name)
+					}
+				}
+			}
+		}
+	}
+
+	return nil
+}
+
+func validateTAC(tac *v1alpha1.TidbClusterAutoScaler) error {
```

Comment:
We need to add some tests for invalid cases. Maybe add TODO here?

Reply:
updated